text/0000-unreservations-2018.md

```diff
@@ -0,0 +1,307 @@
+the standard library, and so we might not want anyone to define these functions,
+not because we will use them ourselves, but because it would be confusing,
+and so the error messages could be improved saying
+*"go look at `std::mem::size_of` instead"*. However, we believe it is better
+to allow users the freedom to use these keywords instead.
+
+# Prior art
+[prior-art]: #prior-art
+
+Not applicable.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+There are none.
+All reservations we will do should be resolved before merging the RFC.
+
+# Appendix
+[appendix]: #appendix
+
+## Reserved keywords we probably don't want to unreserve
+
+The following keywords are used in the nightly compiler and we are sure
+that we want to keep them:
+
+- `yield` - Generators
+- `macro` - Macros 2.0
+
+Additionally, there are known potential use cases / RFCs for:
+
+- `become` - We might want this for guaranteed tail calls.
+  See [the postponed RFC](https://github.com/rust-lang/rfcs/pull/1888).
+
+- `typeof` - We might want this for hypothetical usages such as:
+  ```rust
+  fn foo(x: impl Bar, y: typeof(x)) { .. }
+  ```
+
+- `do` - We might want this for two uses:
+  1. `do { .. } while cond;` loops.
+  2. Haskell style do notation: `let az' = do { x <- ax; y <- ay(x); az };`.
+
+- `abstract` - We might/would like this for:
+  ```rust
+  abstract type Foo: Copy + Debug + .. ;
+  ```
+
+- `override` - This could possibly used for:
+  + OOP inheritance -- unlikely that we'll get such features.
+
+  + specialization -- we do not annotate specialization on the overriding impl
+  but rather say that the base impl is specializable with `default`,
+  wherefore `override` does not make much sense.
+
+  + delegation -- this usage was proposed in the delegations pre-RFC:
+
+    ```rust
+    impl TR for S {
+        delegate * to f;
+
+        #[override(from="f")]
+        fn foo(&self) -> u32 {
+            42
+        }
+    }
+    ```
+
+    which we could rewrite as:
+
+    ```rust
+    impl TR for S {
+        delegate * to f;
+
+        override(from f) fn foo(&self) -> u32 {
+            42
+        }
+    }
+    ```
+
+## Possible future unreservations
+
+### `unsized`
+
+This would be a modifier on types, but we already have `<T: ?Sized>` and we
+could have `T: !Sized` so there seems to be no need for keeping `unsized`.
+
+However, `unsized type` or `unsized struct` might be a desirable syntax for
+declaring a *dynamically sized type (DST)* or completely unsized type.
+Therefore, we will hold of on unreserving `unsized` until we have a better
+ideas of how custom DSTs will work and it's clear we don't need `unsized`
+as a keyword.
+
+### `priv`
+
+Here, `priv` is a privacy / visibility modifier on things like fields, and items.
+An example:
+
+```rust
+priv struct Foo;
+pub struct Bar {
+    priv baz: u8
+}
+```
+
+Since everything is already private by default, `priv` would only be an extra
```

Comment:
Enum variants are not private.

Reply:
Neither are `trait` methods.

Reply:
Two lines down.

> Further, we could possibly use `pub(self)` for `priv` instead.

Reply:
@bjorn3 Thanks; I've corrected this mistake now :)